src/Integration.Vsix/CFamily/PortedFromJava/Analyzer.cs

```diff
@@ -97,8 +97,8 @@ internal class Message : MessagePart
         public bool PartsMakeFlow { get; }
         public MessagePart[] Parts { get; }
 
-        public Message(string ruleKey, string filename, int line, int column, int endLine, int endColumn, string message, bool partsMakeFlow, MessagePart[] parts)
-            : base(filename, line, column, endLine, endColumn, message)
+        public Message(string ruleKey, string filename, int line, int column, int endLine, int endColumn, string text, bool partsMakeFlow, MessagePart[] parts)
+            : base(filename, line, column, endLine, endColumn, text)
```

Comment:
FYI in general, we try to avoid making gratuitous changes to the files that have been ported directly from Java to make it easier to compare our version against the Java version. See the ReadMe.txt in the `PortedfromJava` folder: https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/Integration.Vsix/CFamily/PortedFromJava/ReadMe.txt

As it is, it looks like the Java version has been updated more recently, although a quick look at their changes suggests it's only in the formatting.

Reply:
I made this change because the (response object)[https://github.com/SonarSource/sonarlint-visualstudio/blob/8e01992209d7e136191f64fece181aaff65ee948/src/Integration.Vsix/CFamily/PortedFromJava/Analyzer.cs#L80] contains a "Text" field, but it's not being filled when deserializing because the ctor's argument is named "message".
I can undo this change and instead change the response jsons to have a "Message" field, but it feels awkward. 
Another option is to remove it completely and ignore it in the comparison. Wdyt?

Reply:
Ah. That's an interesting one. The class wasn't designed to be json-serializable and doesn't need to be from a product point of view, but we're now using json-serialization for convenience in our tests.  Hhhmm.

We shouldn't change the public API of the class, since we want it to match the CFamily version.

The risk is we update the class in the future to match any changes made on the CFamily side, and start getting hard-to-diagnose test failures.

As a minimum, I'd add comment to class saying we want it to be json-serializable to simplify testing.
I'd also add a comment to that constructor saying the parameter names have to match the property names for the default json serializer to work.

The next step beyond that would be to write a test that checks it is actually json-serialization-round-trippable, but that might overkill.

Is there anything else we could do to make such failures easy to diagnose?

(for future reference, this is definitely the sort of thing that's worth calling out in a review comment)

Reply:
If the parameter names don't match, the deserialization would not work and the fields would be null. That's the only indicator unfortunately. Perhaps if we encounter this issue in the future then we could invest more into this: e.g., maybe add a TestResponse converts to Response. Or, maybe not use jsons and instead add code functions that return the expected object.

I added a comment for now, let's revise it in the future if it breaks, if it's ok with you.